### PR TITLE
Add PlexonRecordingInterface

### LINKED ISSUE #204: [Feature]: Support Plexon data types
### What would you like to see added to NeuroConv?

We should add support for Plexon data types. Plexon writes data to .plx and .pl2 and offline sorters write data to .ddt and .nex.

* It looks like we already have data in the GIN repo the legacy .plx format and there's already support in neo and SpikeInterface for reading it, so it shouldn't take too much to create a new reader for .plx data
* For the other data types, let's start the process of tracking down test data and adding them to the GIN repo

### Is your feature request related to a problem?

_No response_

### Do you have any interest in helping implement the feature?

No.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/catalystneuro/neuroconv/blob/main/.github/CODE_OF_CONDUCT.rst)
- [X] Have you ensured this bug was not already [reported](https://github.com/catalystneuro/neuroconv/issues)?

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
fix #204 

## Motivation

Adding a PlexonRecordingInterface.

Only the 3rd file in the GIN repo has a recording stream: https://gin.g-node.org/NeuralEnsemble/ephy_testing_data/src/master/plexon

The rest seem to focus on storing the data as waveforms; once Alessio opens up his PR for `write_waveforms` we may want to consider adding `WaveformInterfaces` for the formats that support that data structure.

The header is Intan-like and does not have much extra beyond typical channel properties. Also no session start time.

Example output file: [PlexonRecordingInterface_plexon_recording.zip](https://github.com/catalystneuro/neuroconv/files/10077579/PlexonRecordingInterface_plexon_recording.zip)

## How to test the behavior?

Test on GIN data added.

Have to temporarily disable round-trip SI test since [this line](https://github.com/SpikeInterface/spikeinterface/blob/master/spikeinterface/core/testing.py#L28) in `check_recordings_equal` causes duplicated channel IDs to be specified, which HDF5 does not allow (`TypeError: Indexing elements must be in increasing order`). Have raised [an issue](https://github.com/SpikeInterface/spikeinterface/issues/1092) for this on SI (personal opinion: it's bad input to the `get_traces` call; issue with `check_recordings_equal`, not the `NwbRecordingExtractor`).

## Checklist

- [X] Have you thoroughly read our [Developer Guide](https://neuroconv.readthedocs.io/en/main/developer_guide.html)?
- [X] Have you ensured the PR description clearly describes the problem and solutions?
- [X] Have you checked to ensure that there aren't other open or previously closed [Pull Requests](https://github.com/catalystneuro/neuroconv/pulls) for the same change?
- [X] If this PR fixes an issue, is the first line of the PR description `fix #XXX` where `XXX` is the issue number?
- [X] Did you update the [CHANGLOG.md](https://github.com/catalystneuro/neuroconv/blob/main/CHANGELOG.md) file on your branch to describe your changes?

### CODE REVIEW

File: docs/conversion_examples_gallery/recording/plexon.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
+European Data Format (EDF) conversion
```
Comment (bendichter) [inline docs/conversion_examples_gallery/recording/plexon.rst:1]:
```suggestion
Plexon conversion
```

File: docs/conversion_examples_gallery/recording/plexon.rst [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+-------------------------------------
+
+Install NeuroConv with the additional dependencies necessary for reading EDF data.
```
Comment (bendichter) [inline docs/conversion_examples_gallery/recording/plexon.rst:4]:
```suggestion
Install NeuroConv with the additional dependencies necessary for reading Plexon data.
```

File: docs/conversion_examples_gallery/recording/plexon.rst [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+.. code-block:: bash
+
+    pip install neuroconv[plexon]
+
+Convert edf data to NWB using :py:class:`~neuroconv.datainterfaces.ecephys.plexon.plexondatainterface.PlexonRecordingInterface`.
```
Comment (bendichter) [inline docs/conversion_examples_gallery/recording/plexon.rst:10]:
```suggestion
Convert edf data to NWB using :py:class:`~neuroconv.datainterfaces.ecephys.plexon.plexondatainterface.PlexonRecordingInterface`. Currently, only .plx is supported.
```

File: tests/test_on_data/test_gin_ecephys.py [hunk 4]
```diff
@@ -299,9 +308,14 @@ class TestConverter(NWBConverter):
...
             )
 
-            check_recordings_equal(RX1=recording, RX2=nwb_recording, return_scaled=False)
-            if recording.has_scaled_traces() and nwb_recording.has_scaled_traces():
-                check_recordings_equal(RX1=recording, RX2=nwb_recording, return_scaled=True)
+            # Edge case that only occurs in testing, but should eventually be fixed nonetheless
+            # The NwbRecordingExtractor on spikeinterface experiences an issue when duplicated channel_ids
+            # are specified, which occurs during check_recordings_equal when there is only one channel
+            si_test_ch = [nwb_recording.get_channel_ids()[0], nwb_recording.get_channel_ids()[-1]]
+            if len(si_test_ch) == len(set(si_test_ch)):
```
[the remark this answers is not in the document]
Comment (CodyCBakerPhD) [reply tests/test_on_data/test_gin_ecephys.py:315]: In this specific case, yes - but it's a problem that would exist any time there's a descrepency between these two values. This line will be removed once the fix is available upstream anyhow, so it's all temporary.

Comment (bendichter) [inline tests/test_on_data/test_gin_ecephys.py:315]:
```suggestion
            if nwb_recording.get_channel_ids()[0] != nwb_recording.get_channel_ids()[-1]:
```
ok, how about this?

**Comment by CodyCBakerPhD** [comment]
@bendichter All tests passing except that one little issue with the SI dev branch import that Alessio said he would fix tomorrow

**Review by bendichter — APPROVED** [review]

**Pull request merged**